src/scanner.c

```diff
@@ -0,0 +1,403 @@
+
+#include <tree_sitter/parser.h>
+#include <ctype.h> // isspace()
+#include <string.h> // memcpy()
+
+enum TokenType {
+  NEWLINE,
+  SEMICOLON,
+  ELSE,
+  RAW_STRING_LITERAL,
+  OPEN_PAREN,
+  CLOSE_PAREN,
+  OPEN_BRACE,
+  CLOSE_BRACE,
+  OPEN_BRACKET,
+  CLOSE_BRACKET,
+  OPEN_BRACKET2,
+  CLOSE_BRACKET2
```

Comment:
Note `TOP_LEVEL` has been removed from the token types. We have a separate `Scope` type now that better captures the possible scope states. I think they were kind of conflated with the token type enum previously.